<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="stinekrye" />

<meta name="date" content="2021-10-04" />

<title>5_step_1</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Project_in_Bioinformatics_E21</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stinekrye/Project_in_Bioinformatics_E21">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">5_step_1</h1>
<h4 class="author">stinekrye</h4>
<h4 class="date">2021-10-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Project_in_Bioinformatics_E21/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210919code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210919)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210919code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210919)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C:/Users/ASUS/Google Drev/AU/Aktuelle/PiB/Project_in_Bioinformatics_E21/code/Rhea-master/1.Normalization
</td>
<td style="text-align:left;">
code/Rhea-master/1.Normalization
</td>
</tr>
<tr>
<td style="text-align:left;">
C:/Users/ASUS/Google Drev/AU/Aktuelle/PiB/Project_in_Bioinformatics_E21
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstinekryeProjectinBioinformaticsE21treef18b84e59146e6dae5dc9735e2b5189df6b6cc0btargetblankf18b84ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stinekrye/Project_in_Bioinformatics_E21/tree/f18b84e59146e6dae5dc9735e2b5189df6b6cc0b" target="_blank">f18b84e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstinekryeProjectinBioinformaticsE21treef18b84e59146e6dae5dc9735e2b5189df6b6cc0btargetblankf18b84ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stinekrye/Project_in_Bioinformatics_E21/tree/f18b84e59146e6dae5dc9735e2b5189df6b6cc0b" target="_blank">f18b84e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  code/Rhea-master/
    Untracked:  data/get_seq.r
    Untracked:  oil_spill.nwk
    Untracked:  treemaker.sh

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/5_step_1.Rmd</code>) and HTML (<code>docs/5_step_1.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stinekrye/Project_in_Bioinformatics_E21/blob/f18b84e59146e6dae5dc9735e2b5189df6b6cc0b/analysis/5_step_1.Rmd" target="_blank">f18b84e</a>
</td>
<td>
stinekrye
</td>
<td>
2021-10-06
</td>
<td>
wflow_publish(“./analysis/5_step_1.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stinekrye/Project_in_Bioinformatics_E21/4210d3ae2581ca58dce95094b30f06f836572f95/docs/5_step_1.html" target="_blank">4210d3a</a>
</td>
<td>
stinekrye
</td>
<td>
2021-10-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stinekrye/Project_in_Bioinformatics_E21/blob/059129b9d705f7fe27fcb68c0bcf72acf890dcf2/analysis/5_step_1.Rmd" target="_blank">059129b</a>
</td>
<td>
stinekrye
</td>
<td>
2021-10-06
</td>
<td>
wflow_publish(“./analysis/5_step_1.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stinekrye/Project_in_Bioinformatics_E21/f8883cd6d8e495531fc6012d0d94a28f9651b9b8/docs/5_step_1.html" target="_blank">f8883cd</a>
</td>
<td>
stinekrye
</td>
<td>
2021-10-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stinekrye/Project_in_Bioinformatics_E21/blob/09a18b1ef2a8faac82680460ecf9dea95c944f5e/analysis/5_step_1.Rmd" target="_blank">09a18b1</a>
</td>
<td>
stinekrye
</td>
<td>
2021-10-06
</td>
<td>
wflow_publish(“analysis/5_step_1.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stinekrye/Project_in_Bioinformatics_E21/fec4b0393e0e42efbdcec3c008ef49e5d8c1ff61/docs/5_step_1.html" target="_blank">fec4b03</a>
</td>
<td>
stinekrye
</td>
<td>
2021-10-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stinekrye/Project_in_Bioinformatics_E21/blob/ae8a430b0838781466d1e421eb03955f775bdc05/analysis/5_step_1.Rmd" target="_blank">ae8a430</a>
</td>
<td>
stinekrye
</td>
<td>
2021-10-06
</td>
<td>
wflow_publish(“analysis/5_step_1.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stinekrye/Project_in_Bioinformatics_E21/blob/c8fdc9946779506ece9bed917d3b278b9b128cbf/analysis/5_step_1.Rmd" target="_blank">c8fdc99</a>
</td>
<td>
stinekrye
</td>
<td>
2021-10-06
</td>
<td>
wflow_git_commit(“analysis/5_step_1.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stinekrye/Project_in_Bioinformatics_E21/blob/d8c1e034c9a9c05f4a4e133c985adbff0193e1f7/analysis/5_step_1.Rmd" target="_blank">d8c1e03</a>
</td>
<td>
stinekrye
</td>
<td>
2021-10-06
</td>
<td>
updata files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stinekrye/Project_in_Bioinformatics_E21/blob/e8218b87ae3363fc73c595ee56eedfe799a5aa6e/analysis/5_step_1.Rmd" target="_blank">e8218b8</a>
</td>
<td>
stinekrye
</td>
<td>
2021-10-06
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stinekrye/Project_in_Bioinformatics_E21/blob/5092f3891716de5e1f009209e0673cce7729342f/analysis/5_step_1.Rmd" target="_blank">5092f38</a>
</td>
<td>
stinekrye
</td>
<td>
2021-10-04
</td>
<td>
clean-up of old files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The purpose of this script is to illustrate the workflow of how to produce/explore the following:<br />
- 2.3.2.1: Remove singletons, unwanted sequences and negative controls<br />
- 2.3.2.2: Normalize data and removal of samples with low count (Using Rhea)<br />
- 2.3.3: Alpha diversity (also include pcr data. correlation of index and pcr abundance), (bacteria, chloroplast).</p>
<div id="load-data-and-libraries" class="section level3">
<h3>Load data and libraries</h3>
<pre class="r"><code>library(tidyverse)
library(readxl)
library(phyloseq)
library(vegan)


sampleData &lt;- read_excel(&quot;./data/sampleData.xlsx&quot;) %&gt;% column_to_rownames(., var = &quot;Samples&quot;)
asvTable &lt;- read.delim(&quot;./data/oil_ASVtable.txt&quot;, sep = &quot;\t&quot;, row.names = &quot;ASVs&quot;, check.names = F) #%&gt;% select(., -&#39;row.names&#39;)
taxaId &lt;- read.delim(&quot;./data/oil_ASVtaxid.txt&quot;, sep = &quot;\t&quot;, row.names = &quot;ASVs&quot;) %&gt;% select(., -&#39;row.names&#39;)</code></pre>
</div>
<div id="subset-data-to-include-only-samples-from-gf" class="section level3">
<h3>Subset data to include only samples from GF</h3>
<p>This is done to make the workflow / troubleshooting easier</p>
<ol style="list-style-type: decimal">
<li>Create phyloseq object, which makes subsetting all three files easy.<br />
</li>
<li>Then subset.</li>
</ol>
<pre class="r"><code>asvTable &lt;- otu_table(asvTable, taxa_are_rows = TRUE)
sampleData &lt;- sample_data(sampleData)
taxaId &lt;- tax_table(as.matrix(taxaId))
phyloData &lt;- phyloseq(asvTable,sampleData, taxaId)


gfData &lt;- subset_samples(phyloData, region == &quot;GF&quot;)


GF_asvTable &lt;- as.data.frame(otu_table(gfData))
GF_sampleData &lt;- as.data.frame(sample_data(gfData))
GF_taxaId &lt;- as.data.frame(tax_table(taxaId))</code></pre>
</div>
<div id="create-the-input-file-for-rhea" class="section level3">
<h3>Create the input file for Rhea</h3>
<p>The table needed for Rhea is a tab seperated OTU/ASV table with ASVs as rownames and samplenames as columns. In addidtion to that, we need the taxonomy levels seperated with semicolons as the last column names "taxonomy.</p>
<p>oil_ASVtable needs a taxonomy column.</p>
<pre class="r"><code>taxo &lt;- GF_taxaId
taxo$taxonomy &lt;- paste(taxo$Kingdom, taxo$Phylum, taxo$Order, taxo$Family, taxo$Class, taxo$Genus, taxo$V7, sep = &quot;;&quot;) # concatenate rows
taxo &lt;- taxo %&gt;% select(c(taxonomy))                                                                                   # make dataframe ready for merge
rhea_GF_asvTable &lt;- merge(as.data.frame(GF_asvTable), taxo, by = 0)                                                    # 0 = rownames
rhea_GF_asvTable &lt;- column_to_rownames(rhea_GF_asvTable, var = &quot;Row.names&quot;)                                            # restore rownames
#write.table(rhea_GF_asvTable, file = &quot;oil_ASVtable_RHEA.txt&quot;, sep = &quot;\t&quot;, quote = F) # Is moved to the Rhea folder
head(rhea_GF_asvTable)[123:128]</code></pre>
<pre><code>         FGJuly_116_S106_F_filt.fastq.gz FGJuly_141_S118_F_filt.fastq.gz
ASV1                                 743                            3237
ASV10                                148                              25
ASV100                                 0                               0
ASV1000                                0                               0
ASV10000                               0                               0
ASV10001                               0                               0
         FGJuly_143_S35_F_filt.fastq.gz FGJuly_157_S47_F_filt.fastq.gz
ASV1                               3485                            227
ASV10                                 0                            213
ASV100                                0                              0
ASV1000                               0                              0
ASV10000                              0                              0
ASV10001                              0                              0
         FGJuly_158_S59_F_filt.fastq.gz FGJuly_159_S71_F_filt.fastq.gz
ASV1                                175                             19
ASV10                               154                            418
ASV100                                0                              0
ASV1000                               0                              0
ASV10000                              0                              0
ASV10001                              0                              0</code></pre>
</div>
</div>
<div id="remove-singletons-unwanted-sequences-and-negative-controls" class="section level2">
<h2>2.3.2.1 Remove singletons, unwanted sequences and negative controls</h2>
<p>First we need to remove singletons, chloroplast and mitochondrial species, and negatibe controls. - Ioannis have removed singletons, chloroplast and mitocondrial species - We have no negative controls in the in situ samples. By negative controls I mean a control which has not been in contact with any bacteria, but has been sequences to investigate the amount of “background noise” from the project.</p>
</div>
<div id="normalization-and-removal-of-samples-with-low-count" class="section level2">
<h2>2.3.2.2 Normalization and removal of samples with low count</h2>
<div id="rhea-workflow" class="section level3">
<h3>Rhea Workflow</h3>
<p>Run two times:<br />
- First time to identify underrepresented samples, which has to be removed<br />
- Second time the analysis is made based on the updated OTU table.</p>
<div id="first-rhea-run" class="section level4">
<h4>First Rhea run</h4>
<p>The following has been changed in the Rhea Normalization.R file: setwd(“C:/Users/ASUS/Google Drev/AU/Aktuelle/PiB/Project_in_Bioinformatics_E21/code/Rhea-master/1.Normalization”) file_name &lt;- “oil_ASVtable_RHEA.txt” labelCutoff &lt;- 10</p>
<pre class="r"><code>#source(&quot;./code/Rhea-master/1.Normalization/Normalization.R&quot;)
#setwd(&quot;C:/Users/ASUS/Google Drev/AU/Aktuelle/PiB/Project_in_Bioinformatics_E21&quot;)</code></pre>
<p>Use the slope of the rarefaction curve to decide cut-off<br />
- ** How do you do this? **<br />
- My idea: Make a plot whuch enables us to find the outliers and cut them away. The cut-off can be choosen based on a visual expection</p>
<pre class="r"><code>rslope &lt;- read.delim(&quot;./code/Rhea-master/1.Normalization/1_RarefactionCurve.TAB&quot;, sep = &quot;\t&quot;, check.names = F)

rslope %&gt;% ggplot(mapping = aes(x = reorder(SampleID, -slope), y = slope)) +
  geom_point() +
  ylim(0, 2) +
  theme_classic()+
  theme(axis.text.x=element_blank())+
  xlab(&quot;SampleID (reordered)&quot;) +
  ylab(&quot;Slope of rarefaction curve&quot;)</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/5_step_1.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>First find the names of the samples, which gets excluded based on a large slope</p>
<pre class="r"><code># set threshold set by visual inspection of the plot
cut &lt;- rslope[rslope$slope &gt; 0.75,]</code></pre>
<p>Then find samples, which gets excluded based on a low number of total reads. This threshold cuts away samples with a size of 0-3. The lowest read counts after removal of those is just below 20000.</p>
<pre class="r"><code># Cut away samples based on too low count.
too_low_sample_size &lt;- colnames(GF_asvTable)[colSums(GF_asvTable) &lt; 10000]</code></pre>
<p>Remove those samples from the phyloseq object, so we can use it to generate a new input file for Rhea</p>
<pre class="r"><code># Cut away unwanted sequences
unwanted &lt;- c(cut$SampleID, too_low_sample_size)
wanted &lt;- colnames(GF_asvTable)[!colnames(GF_asvTable) %in% unwanted]
gfDataCut &lt;- prune_samples(wanted,gfData)</code></pre>
<p>Make new Rhea file</p>
<pre class="r"><code>taxo &lt;- as.data.frame(tax_table(taxaId))
taxo$taxonomy &lt;- paste(taxo$Kingdom, taxo$Phylum, taxo$Order, taxo$Family, taxo$Class, taxo$Genus, taxo$V7, sep = &quot;;&quot;) # concatenate rows
taxo &lt;- taxo %&gt;% select(c(taxonomy))                                                                                   # make dataframe ready for merge
rhea_GF_asvTable &lt;- merge(as.data.frame(otu_table(gfDataCut)), taxo, by = 0)                                                    # 0 = rownames
rhea_GF_asvTable &lt;- column_to_rownames(rhea_GF_asvTable, var = &quot;Row.names&quot;)                                            # restore rownames
#write.table(rhea_GF_asvTable, file = &quot;oil_ASVtable_RHEA_CUT.txt&quot;, sep = &quot;\t&quot;, quote = F) # Is moved to the Rhea folder</code></pre>
</div>
<div id="second-rhea-run" class="section level4">
<h4>Second Rhea run</h4>
<p>The following has been changed in the Rhea Normalization.R file: setwd(“C:/Users/ASUS/Google Drev/AU/Aktuelle/PiB/Project_in_Bioinformatics_E21/code/Rhea-master/1.Normalization”) file_name &lt;- “oil_ASVtable_RHEA_CUT.txt” labelCutoff &lt;- 5</p>
<pre class="r"><code>#source(&quot;../code/Rhea-master/1.Normalization/Normalization.R&quot;)
#setwd(&quot;C:/Users/ASUS/Google Drev/AU/Aktuelle/PiB/Project_in_Bioinformatics_E21&quot;)</code></pre>
<p>The output file with normalized read counts is used in the next step</p>
</div>
</div>
</div>
<div id="calculate-alpha-diversity" class="section level2">
<h2>2.3.3.1 Calculate Alpha diversity</h2>
<div id="follow-the-rhea-workflow" class="section level3">
<h3>Follow the Rhea workflow</h3>
<p>Use the normalized read counts</p>
<pre class="r"><code>#source(&quot;../code/Rhea-master/2.Alpha-Diversity/Alpha-Diversity.R&quot;)
#setwd(&quot;C:/Users/ASUS/Google Drev/AU/Aktuelle/PiB/Project_in_Bioinformatics_E21&quot;)</code></pre>
</div>
<div id="visualization-of-alpha-diversity" class="section level3">
<h3>Visualization of alpha diversity</h3>
<p>Idea: Make two plots: - One with qPCR abundance values against both of the effective indexes (color coded) - One with one of the effective indexes against the sample name. The point has to be colored in accordance to some grouping Eg. (1st grouping: in situ + sea water, 2nd grouping: Deep, sediment, surface)</p>
<p>Merge the sampleData and the alpha-diversity measures in one table to make plotting easier</p>
<pre class="r"><code>alpha &lt;- read.delim(&quot;./code/Rhea-master/2.Alpha-Diversity/alpha-diversity.TAB&quot;, sep = &quot;\t&quot;, check.names = F, row.names = 1)
alphaSampleData &lt;- merge(sampleData, alpha, by = 0)</code></pre>
<div id="plot-type-1-show-richness-in-different-groups" class="section level4">
<h4>Plot type 1: Show richness in different groups</h4>
<p>The data is seperated into in situ vs. SW and then additionally seperated into deep, sediment and surface</p>
<pre class="r"><code>alphaSampleData %&gt;% ggplot(mapping = aes(x = location, y = Simpson.Effective)) +
  geom_violin() +
  facet_wrap(&quot;type&quot;)</code></pre>
<p><img src="figure/5_step_1.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>alphaSampleData %&gt;% ggplot(mapping = aes(x = location, y = Simpson.Effective)) +
  geom_jitter(aes(color = oil)) +
  facet_wrap(&quot;type&quot;)</code></pre>
<p><img src="figure/5_step_1.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>alphaSampleData %&gt;% ggplot(mapping = aes(x = location, y = Simpson.Effective)) +
  geom_jitter(aes(color = Evenness)) +
  facet_wrap(&quot;type&quot;)</code></pre>
<p><img src="figure/5_step_1.Rmd/unnamed-chunk-13-3.png" width="672" style="display: block; margin: auto;" /></p>
<blockquote>
<p>How can it make sense that the blanks has high alpha diversity, when Shannon and Simpson index is used?</p>
</blockquote>
<p>When using effective richness (enumeration) the distribution looks random. Does this show that the samples with oils might be dominated by a few species?</p>
<ul>
<li>Notes on Simpson: Look up the formula. Takes the number of species and their abundance into account. A sample with large evenness is considered more diverse than a sample dominated by a few species (possible enrichment)</li>
</ul>
<p>Notes on Shannon: H = (Relative proportion of species * the natural logarithmic function of that relative proportion) summed over all species. Evenness is H / ln(S) where S is the total number of species in that sample. Low evenness means that a few species dominate *</p>
</div>
<div id="plot-type-2-visualize-the-relationsship-between-alpha-diversity-and-cpcr-numbers" class="section level4">
<h4>Plot type 2: Visualize the relationsship between alpha-diversity and cPCR numbers</h4>
<p>The y axis is log scale. This seperates the points better.</p>
<blockquote>
<p>Is it bad to have a correlation? Could that mean a correlation between the size of the sample and the richness? If we have no correlation, does that mean that the effective richness does not depend so much on sample size?</p>
</blockquote>
<pre class="r"><code>alphaSampleData %&gt;% ggplot(mapping = aes(y = QPCRCopies)) + 
  #geom_point(aes(x = Shannon.Effective, color = &quot;Shannon&quot;)) +
  geom_point(aes(x = Simpson.Effective, color = &quot;Simpson&quot;)) +
  scale_y_log10() +
  #xlim(0,100) +
  NULL</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/5_step_1.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<blockquote>
<p>Maybe do the same with phyloseq?</p>
</blockquote>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=Danish_Denmark.1252  LC_CTYPE=Danish_Denmark.1252   
[3] LC_MONETARY=Danish_Denmark.1252 LC_NUMERIC=C                   
[5] LC_TIME=Danish_Denmark.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] vegan_2.5-7     lattice_0.20-44 permute_0.9-5   phyloseq_1.36.0
 [5] readxl_1.3.1    forcats_0.5.1   stringr_1.4.0   dplyr_1.0.7    
 [9] purrr_0.3.4     readr_2.0.1     tidyr_1.1.3     tibble_3.1.4   
[13] ggplot2_3.3.5   tidyverse_1.3.1 workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] nlme_3.1-152           bitops_1.0-7           fs_1.5.0              
 [4] lubridate_1.7.10       httr_1.4.2             rprojroot_2.0.2       
 [7] GenomeInfoDb_1.28.4    tools_4.1.1            backports_1.2.1       
[10] utf8_1.2.2             R6_2.5.1               mgcv_1.8-36           
[13] DBI_1.1.1              BiocGenerics_0.38.0    colorspace_2.0-2      
[16] rhdf5filters_1.4.0     ade4_1.7-18            withr_2.4.2           
[19] tidyselect_1.1.1       compiler_4.1.1         git2r_0.28.0          
[22] cli_3.0.1              rvest_1.0.1            Biobase_2.52.0        
[25] xml2_1.3.2             labeling_0.4.2         scales_1.1.1          
[28] digest_0.6.27          rmarkdown_2.11         XVector_0.32.0        
[31] pkgconfig_2.0.3        htmltools_0.5.2        dbplyr_2.1.1          
[34] fastmap_1.1.0          highr_0.9              rlang_0.4.11          
[37] rstudioapi_0.13        farver_2.1.0           jquerylib_0.1.4       
[40] generics_0.1.0         jsonlite_1.7.2         RCurl_1.98-1.5        
[43] magrittr_2.0.1         GenomeInfoDbData_1.2.6 Matrix_1.3-4          
[46] biomformat_1.20.0      Rhdf5lib_1.14.2        Rcpp_1.0.7            
[49] munsell_0.5.0          S4Vectors_0.30.0       fansi_0.5.0           
[52] ape_5.5                lifecycle_1.0.0        stringi_1.7.4         
[55] whisker_0.4            yaml_2.2.1             MASS_7.3-54           
[58] zlibbioc_1.38.0        rhdf5_2.36.0           plyr_1.8.6            
[61] grid_4.1.1             parallel_4.1.1         promises_1.2.0.1      
[64] crayon_1.4.1           splines_4.1.1          Biostrings_2.60.2     
[67] haven_2.4.3            multtest_2.48.0        hms_1.1.0             
[70] knitr_1.34             pillar_1.6.2           igraph_1.2.6          
[73] reshape2_1.4.4         codetools_0.2-18       stats4_4.1.1          
[76] reprex_2.0.1           glue_1.4.2             evaluate_0.14         
[79] data.table_1.14.0      modelr_0.1.8           vctrs_0.3.8           
[82] tzdb_0.1.2             httpuv_1.6.3           foreach_1.5.1         
[85] cellranger_1.1.0       gtable_0.3.0           assertthat_0.2.1      
[88] xfun_0.26              broom_0.7.9            later_1.3.0           
[91] survival_3.2-11        iterators_1.0.13       IRanges_2.26.0        
[94] cluster_2.1.2          ellipsis_0.3.2        </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
